cluster/ec2_banzaicloud.go

```diff
@@ -0,0 +1,504 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	pipConfig "github.com/banzaicloud/pipeline/config"
+	"github.com/banzaicloud/pipeline/internal/cluster"
+	banzaicloudDB "github.com/banzaicloud/pipeline/internal/providers/banzaicloud"
+	"github.com/banzaicloud/pipeline/model"
+	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
+	"github.com/banzaicloud/pipeline/pkg/cluster/banzaicloud"
+	"github.com/banzaicloud/pipeline/pkg/common"
+	pkgError "github.com/banzaicloud/pipeline/pkg/errors"
+	"github.com/banzaicloud/pipeline/secret"
+	"github.com/goph/emperror"
+	"github.com/jinzhu/gorm"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+)
+
+var _ CommonCluster = (*EC2ClusterBanzaiCloudDistribution)(nil)
+
+type EC2ClusterBanzaiCloudDistribution struct {
+	db    *gorm.DB
+	model *banzaicloudDB.EC2BanzaiCloudClusterModel
+	//amazonCluster *ec2.EC2 //Don't use this directly
+	APIEndpoint string
+	CommonClusterBase
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetSecurityScan() bool {
+	return c.model.Cluster.SecurityScan
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) SetSecurityScan(scan bool) {
+	c.model.Cluster.SecurityScan = scan
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetLogging() bool {
+	return c.model.Cluster.Logging
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) SetLogging(l bool) {
+	c.model.Cluster.Logging = l
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetMonitoring() bool {
+	return c.model.Cluster.Monitoring
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) SetMonitoring(m bool) {
+	c.model.Cluster.Monitoring = m
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetID() uint {
+	return c.model.Cluster.ID
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetUID() string {
+	return c.model.Cluster.UID
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetOrganizationId() uint {
+	return c.model.Cluster.OrganizationID
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetName() string {
+	return c.model.Cluster.Name
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetCloud() string {
+	return c.model.Cluster.Cloud
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetDistribution() string {
+	return c.model.Cluster.Distribution
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetLocation() string {
+	return c.model.Cluster.Location
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetCreatedBy() uint {
+	return c.model.Cluster.CreatedBy
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetSecretId() string {
+	return c.model.Cluster.SecretID
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) GetSshSecretId() string {
+	return c.model.Cluster.SSHSecretID
+}
+
+func (c *EC2ClusterBanzaiCloudDistribution) SaveSshSecretId(sshSecretId string) error {
+	c.model.Cluster.SSHSecretID = sshSecretId
+
+	err := c.db.Save(&c.model).Error
+	if err != nil {
+		return errors.Wrap(err, "failed to save ssh secret id")
```

Comment:
Because the logger lacks the context in this package please use emperor.Wrapwith and add context to the error.